### PR TITLE
Add missing RoBERTa converter

### PR DESCRIPTION
**Pull request by Ahmed0830**
## Description of the change

Split out of #2929 per review feedback, isolating the RoBERTa HF→KerasHub converter so it can be reviewed and numerically verified independently of the other converters in that PR.

- Adds `keras_hub/src/utils/transformers/convert_roberta.py` and wires `model_type == "roberta"` into `TransformersPresetLoader`.
- Adds `convert_roberta_test.py` (`test_convert_tiny_preset`,  `test_class_detection`).
- Additionally addresses the reviewer's follow-up: `tools/checkpoint_conversion/convert_roberta_checkpoints.py` previously hand-ported weights from the original *fairseq* RoBERTa checkpoint, layer by layer. Since `convert_roberta.py` now handles the Hugging Face weight mapping directly via `from_preset()`, that manual porting code is not needed, the script now loads straight from huggingface via `from_preset()`, matching the pattern in `convert_bert_sentence_transformer_checkpoints.py`.

## Verification

Ran the rewritten `convert_roberta_checkpoints.py --preset roberta_base_en` end-to-end against the live `FacebookAI/roberta-base` checkpoint and compared `last_hidden_state` against the reference HF model.
**Colab Reference:** [colab](https://gist.github.com/Ahmed0830/dcede0244002fd9674f5c9475aa4f8d4)

- Resulting preset directory loads correctly (config.json, model.weights.h5, tokenizer.json, assets/).
- `pytest --run_extra_large keras_hub/src/utils/transformers/convert_roberta_test.py` — 2 passed.

## Checklist

- [x] I have added all the necessary unit tests for my change.
- [x] I have verified that my change does not break existing code and works with all backends (TensorFlow, JAX, and PyTorch).
- [x] My PR is based on the latest changes of the main branch (if unsure, rebase the code).
- [x] I have followed the Keras Hub [Model contribution guidelines](https://github.com/keras-team/keras-hub/blob/master/CONTRIBUTING_MODELS.md) in making these changes.
- [x] I have followed the Keras Hub [API design guidelines](https://github.com/keras-team/keras-hub/blob/master/API_DESIGN_GUIDE.md) in making these changes.
- [x] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).

### CODE REVIEW

File: tools/checkpoint_conversion/convert_roberta_checkpoints.py [hunk 1]
```diff
@@ -304,31 +56,33 @@ def check_output(
     print("HF output:", hf_output[0, 0, :10])
     print("Difference:", np.mean(keras_hub_output - hf_output.detach().numpy()))
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply tools/checkpoint_conversion/convert_roberta_checkpoints.py]: Fixed. check_output now slices to real (non-padding) tokens via padding_mask and asserts np.testing.assert_allclose(atol=1e-4, rtol=1e-4) instead of printing a whole-sequence mean.

File: keras_hub/src/utils/transformers/convert_roberta.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        loader.port_weight(
+            keras_variable=encoder_layer._self_attention_layer.output_dense.kernel,
+            hf_weight_key=f"{hf_prefix}.attention.output.dense.weight",
+            hook_fn=transpose_and_reshape,
+        )
+        loader.port_weight(
+            keras_variable=encoder_layer._self_attention_layer.output_dense.bias,
+            hf_weight_key=f"{hf_prefix}.attention.output.dense.bias",
+            hook_fn=lambda hf_tensor, shape: np.reshape(hf_tensor, shape),
+        )
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/utils/transformers/convert_roberta.py:95]: query_dense/key_dense/value_dense/output_dense are public property aliases for the private attrs in Keras's MultiHeadAttention, so the current code is correct.The tests already pass without an AttributeError.

File: keras_hub/src/utils/transformers/convert_roberta_test.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from keras_hub.src.models.text_classifier import TextClassifier
+from keras_hub.src.tests.test_case import TestCase
+
+
+class TestRobertaConverter(TestCase):
+    @pytest.mark.extra_large
+    def test_convert_tiny_preset(self):
+        model = RobertaTextClassifier.from_preset(
+            "hf://FacebookAI/roberta-base", num_classes=2
+        )
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/utils/transformers/convert_roberta_test.py:17]: the model size is under 500 mb, it should be fine. also tried the suggested preset anyways, there was an error.

**Review by laxmareddyp — APPROVED** [review]
Thanks..!LGTM
